### PR TITLE
feat: Add Reverse Shell feature with web interface and menu integration

### PR DESCRIPTION
**Pull request by 9dl**
#### Proposed Changes ####  
Reverse Shell feature with Web Interface for controlling and executing commands remotely through BruceC2.

#### Types of Changes ####  
- New Feature

#### Verification ####  
- Verify that the reverse shell connects to the machine and allows for remote execution of commands via the BruceC2 interface.
- Ensure the web interface works at `http://192.168.4.1` and displays the status and interaction controls.

#### Testing ####  
- Not covered by existing unit tests.
- Consider adding integration tests to ensure the connection and command execution work properly across different platforms.

#### Linked Issues ####  
- N/A (If there are any related issues, they can be linked here)

#### User-Facing Change ####  
```release-note
- Introduced a Reverse Shell feature with web interface for remote command execution via BruceC2.
- Provides TCP-based shell access to a machine and a web interface to manage commands.
```

#### Further Comments ####  
- To execute, start **Reverse Shell Mode** in Bruce and wait for BruceC2 to establish the connection.
- It’s recommended to use **BadUSB scripts** to automate the execution of BruceC2 on the target machine.

```cpp// Developed by Fourier (github.com/9dl)
// Use BruceC2 to interact with the reverse shell server
// BruceC2: https://github.com/9dl/Bruce-C2
// To use BruceC2:
// 1. Start Reverse Shell Mode in Bruce
// 2. Start BruceC2 and wait.
// 3. Visit 192.168.4.1 in your browser to access the web interface for shell executing.

// 32bit: https://github.com/9dl/Bruce-C2/releases/download/v1.0/BruceC2_windows_386.exe
// 64bit: https://github.com/9dl/Bruce-C2/releases/download/v1.0/BruceC2_windows_amd64.exe```

### CODE REVIEW

**Comment by 9dl** [comment]
image
image

**Comment by bmorcelli** [comment]
Can you rebase to fix the conflicts?

**Review by bmorcelli — CHANGES_REQUESTED** [review]
You have all your code in the /core/menu_items files.. it is not how things work...

please creeate new set of files ( ReverseShell.h/cpp) in /modules/wifi/ and add the call for this feature into the Wifi menu.. doesn't make sense have an access to a Wifi application in the Main menu..

**Comment by 9dl** [comment]
I will fix it when I'm home.

File: src/core/menu_items/WifiMenu.cpp [hunk 1]
```diff
@@ -14,6 +14,18 @@
...
+#include "modules/reverseShell/reverseShell.h"
+// Developed by Fourier (github.com/9dl)
+// Use BruceC2 to interact with the reverse shell server
+// BruceC2: https://github.com/9dl/Bruce-C2
+// To use BruceC2:
+// 1. Start Reverse Shell Mode in Bruce
+// 2. Start BruceC2 and wait.
+// 3. Visit 192.168.4.1 in your browser to access the web interface for shell executing.
+
+// 32bit: https://github.com/9dl/Bruce-C2/releases/download/v1.0/BruceC2_windows_386.exe
```
Comment (bmorcelli) [inline src/core/menu_items/WifiMenu.cpp:26]: Please Add these files into a .zip in the ./sd_files/reverseshell folder and, if you may, add the BADUSB payload to this folder, so we can ,merge with all files set..

Edit: I am talking about the .exe programs into a .zip..

**Pull request merged**